common/auto-updater/http-downloader.cpp

```diff
@@ -0,0 +1,186 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+
+
+#ifdef ENABLE_RS_AUTO_UPDATER
+#include <sstream>
+#include <iostream>
+#include <thread>
+#include <atomic>
+#include <mutex>
+#include <curl/curl.h>
+#include <curl/easy.h>
+#include "types.h"
+#endif // ENABLE_RS_AUTO_UPDATER
+
+#include "http-downloader.h"
+
+
+namespace rs2
+{
+#ifndef ENABLE_RS_AUTO_UPDATER
+    // Dummy functions
+    http_downloader::http_downloader() {};
+    http_downloader::~http_downloader() {};
+    bool http_downloader::download_to_stream(const std::string& url, std::stringstream &output, user_callback_func_type user_callback_func) { return false; }
+    bool http_downloader::download_to_file(const std::string& url, const std::string &file_name, user_callback_func_type user_callback_func) { return false; }
+    bool http_downloader::download_to_bytes_vector(const std::string& url, std::vector<uint8_t> &output, user_callback_func_type user_callback_func) { return false; }
+
+#else
+
+    std::mutex initialize_mutex;
+    static const curl_off_t HALF_SEC = 500000; // User call back function delay
+    static const int CONNECT_TIMEOUT = 5L; // Libcurl download timeout 5 [Sec]
+
+    struct progress_data {
+        curl_off_t last_run_time;
+        user_callback_func_type user_callback_func;
+        CURL *curl;
+    };
+
+
+    size_t stream_write_callback(void *ptr, size_t size, size_t nmemb, void *stream) {
```

Comment:
Pls use full names for variables. It is not clear what ptr and nmemb stand for

Reply:
Opening braces should be on their own lines, too

Reply:
Will be fixed

Reply:
Done